src/server.js

```diff
@@ -85,6 +86,18 @@ app
     // This endpoint handles OAuth2 requests (exchanges code for token)
     server.get("/callback", passport.authenticate("oauth2"), (req, res) => {
       // After success, redirect to the page we came from originally
+      // Before redirecting, we add client config info in cookie. They're needed when refreshing a token
+      const clientId = process.env.OAUTH2_CLIENT_ID;
+      const clientSecret = process.env.OAUTH2_CLIENT_SECRET;
+      const refreshTokenUrl = process.env.OAUTH2_IDP_REFRESH_TOKEN_URL;
+      let encoded = "";
+      try {
+        const str = JSON.stringify({ clientId, clientSecret, refreshTokenUrl });
+        encoded = Buffer.from(str).toString("base64");
+      } catch (error) {
+        logger.error("Error creating encoded config string", error);
+      }
+      res.cookie("oauth_client_config", encoded);
```

Comment:
Why is a cookie needed here? Can you add these three config values to the public app config instead? `publicRuntimeConfig` in `next.config.js`